src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/AbstractSamplerCreator.java

```diff
@@ -86,30 +75,49 @@ public AbstractSamplerCreator() {
     /**
      * @return int request number
      */
-    protected static int getRequestNumber() {
+    public int getRequestNumber() {
```

Comment:
Why did you change it to be non-static? (I am not sure, I like the static, but I would like to here your reasoning.

Reply:
The method need to be public not protected (same package)
This method return value from a static value but i don't see why this method need to be static.
Other methods are not static.